.gitignore

```diff
@@ -1,2 +1,3 @@
-vendor
-composer.lock
+/vendor/
+/composer.lock
```

Comment:
- [`php-cs-fixer` is an application](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/composer.json#L3) but does not lock deps.
- [`composer/composer`](https://github.com/composer/composer/blob/master/composer.json#L6) is a library despite the fact I consider it a project/application according to the criteria mentioned above.
- [`phpunit`](https://github.com/sebastianbergmann/phpunit/blob/5.5/composer.json#L4) is a library. But that's fine to me because it's a tool and a unit testing framework you use in your codebase. It does not lock deps.

It actually seems there is no strong consensus on what is a project, an application, a tool, or a library in the PHP community. Let's not have one here :trollface: